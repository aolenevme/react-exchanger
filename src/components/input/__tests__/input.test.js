import React from "react";
import renderer from "react-test-renderer";
import constant from "lodash/constant.js";

import Input from "../input.js";

describe("<Input />", () => {
    it("is rendered correctly", () => {
        expect(renderer
            .create(<Input prefix={constant("+")} value="2020"/>)
            .toJSON()).toMatchSnapshot();

        expect(renderer
            .create(<Input isDisabled/>)
            .toJSON()).toMatchSnapshot();

        expect(renderer
            .create(<Input isDisabled={false}/>)
            .toJSON()).toMatchSnapshot();

        expect(renderer
            .create(<Input isDisabled value="2020"/>)
            .toJSON()).toMatchSnapshot();

        expect(renderer
            .create(<Input isDisabled={false} value="2020"/>)
            .toJSON()).toMatchSnapshot();

        expect(renderer
            .create(<Input isDisabled={false} value="0"/>)
            .toJSON()).toMatchSnapshot();
    });

    describe("events", () => {
        it("reacts to onInput", () => {
            const onInput = jest.fn();

            const tree = renderer
                .create(<Input onInput={onInput}/>)
                .toJSON();

            tree.children[1].props.onInput();

            expect(onInput).toHaveBeenCalled();
        });
    });
});
