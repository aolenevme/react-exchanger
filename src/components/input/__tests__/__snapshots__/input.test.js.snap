// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> is rendered correctly 1`] = `
.c0 {
  font-size: 3rem;
  line-height: 3rem;
  color: #f3f6ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0;
  min-width: 3rem;
  height: 3rem;
  border: 0 solid transparent;
  border-right: 1px solid transparent;
  background-color: transparent;
  cursor: pointer;
}

.c0:focus-within {
  -webkit-animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
  animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
}

.c1 {
  margin: 0;
  padding: 0;
  width: 0;
  height: 0;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-outer-spin-button,
.c1::-webkit-inner-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1:invalid {
  box-shadow: none;
}

<label
  className="main-text__MainText-c4b6rt-0 c0"
>
  +
  2020
  <input
    className="c1"
    disabled={false}
    onInput={[Function]}
    type="number"
    value="2020"
  />
</label>
`;

exports[`<Input /> is rendered correctly 2`] = `
.c0 {
  font-size: 3rem;
  line-height: 3rem;
  color: #f3f6ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0;
  min-width: 3rem;
  height: 3rem;
  border: 0 solid transparent;
  border-right: 1px solid transparent;
  background-color: transparent;
  cursor: initial;
}

.c0:focus-within {
  -webkit-animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
  animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
}

.c1 {
  margin: 0;
  padding: 0;
  width: 0;
  height: 0;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-outer-spin-button,
.c1::-webkit-inner-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1:invalid {
  box-shadow: none;
}

<label
  className="main-text__MainText-c4b6rt-0 c0"
>
  
  <input
    className="c1"
    disabled={true}
    onInput={[Function]}
    type="number"
    value=""
  />
</label>
`;

exports[`<Input /> is rendered correctly 3`] = `
.c0 {
  font-size: 3rem;
  line-height: 3rem;
  color: #f3f6ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0;
  min-width: 3rem;
  height: 3rem;
  border: 0 solid transparent;
  border-right: 1px solid transparent;
  background-color: transparent;
  cursor: pointer;
}

.c0:focus-within {
  -webkit-animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
  animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
}

.c1 {
  margin: 0;
  padding: 0;
  width: 0;
  height: 0;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-outer-spin-button,
.c1::-webkit-inner-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1:invalid {
  box-shadow: none;
}

<label
  className="main-text__MainText-c4b6rt-0 c0"
>
  
  <input
    className="c1"
    disabled={false}
    onInput={[Function]}
    type="number"
    value=""
  />
</label>
`;

exports[`<Input /> is rendered correctly 4`] = `
.c0 {
  font-size: 3rem;
  line-height: 3rem;
  color: #f3f6ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0;
  min-width: 3rem;
  height: 3rem;
  border: 0 solid transparent;
  border-right: 1px solid transparent;
  background-color: transparent;
  cursor: initial;
}

.c0:focus-within {
  -webkit-animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
  animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
}

.c1 {
  margin: 0;
  padding: 0;
  width: 0;
  height: 0;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-outer-spin-button,
.c1::-webkit-inner-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1:invalid {
  box-shadow: none;
}

<label
  className="main-text__MainText-c4b6rt-0 c0"
>
  2020
  <input
    className="c1"
    disabled={true}
    onInput={[Function]}
    type="number"
    value="2020"
  />
</label>
`;

exports[`<Input /> is rendered correctly 5`] = `
.c0 {
  font-size: 3rem;
  line-height: 3rem;
  color: #f3f6ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0;
  min-width: 3rem;
  height: 3rem;
  border: 0 solid transparent;
  border-right: 1px solid transparent;
  background-color: transparent;
  cursor: pointer;
}

.c0:focus-within {
  -webkit-animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
  animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
}

.c1 {
  margin: 0;
  padding: 0;
  width: 0;
  height: 0;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-outer-spin-button,
.c1::-webkit-inner-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1:invalid {
  box-shadow: none;
}

<label
  className="main-text__MainText-c4b6rt-0 c0"
>
  2020
  <input
    className="c1"
    disabled={false}
    onInput={[Function]}
    type="number"
    value="2020"
  />
</label>
`;
